src/transforms/remap.rs

```diff
@@ -54,16 +55,20 @@ impl TransformConfig for RemapConfig {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Clone, Debug)]
 pub struct Remap {
     program: Program,
     timezone: TimeZone,
     drop_on_error: bool,
     drop_on_abort: bool,
+    enrichment_tables: enrichment::TableSearch,
 }
 
 impl Remap {
-    pub fn new(config: RemapConfig) -> crate::Result<Self> {
+    pub fn new(
+        config: RemapConfig,
+        enrichment_tables: &enrichment::TableRegistry,
+    ) -> crate::Result<Self> {
```

Comment:
nit: I think it would be cleaner if we kept the old `pub fn new` impl with a default (empty) read table, and add a new (private) function `new_with_enrichment_tables` which is only called within this module itself.

It keeps the option to inject custom enrichment tables scoped to this module (where it belongs), and reduces the diff of the PR since there's no requirement to update all places `Remap::new` is called.

We can tackle this in a follow-up PR once this is shipped.